src/lib/support/ZclString.h

```diff
@@ -0,0 +1,36 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+namespace chip {
+
+/**
+ * @brief Create ZCL string from char
+ */
+static uint8_t * MakeZclCharString(uint8_t * zclString, const char * cString)
+{
+    size_t len = strlen(cString);
+    if (len > 254)
+    {
+        len = 0;
+    }
+    zclString[0] = static_cast<uint8_t>(len);
+    memcpy(&zclString[1], cString, zclString[0]);
```

Comment:
How do we know this is safe?  Nothing here guarantees `zclString` has 255 bytes of space in it (which is all we know as far as limits on data sizes go), and in fact a bunch of callsites don't have that much space...

Reply:
zclString is now a pointer so it's up to a consumer of a function to allocate the memory for the variable.

Reply:
My point is that we should not be doing that.  We should be passing in how much buffer space the consumer actually has.  Ideally by making zclString be a `MutableByteSpan`, but worst-case by having a separate argument for how much buffer we actually have.